scss/functions/_get-color.scss

```diff
@@ -1,10 +1,31 @@
 @use '../variables' as *;
 @use 'sass:map';
 
+// @function get-color($type, $level) {
+//   // add the logic here
+//   // use if else to check if the type and level are valid
+//   // if they are valid, return the color
+//   // if they are not valid warn 'Invalid color type or level' and return null
+//   // null is a special value in Sass that means nothing, and it is used to represent the absence of a value, you need to return null if the type or level is not valid
+//
+//prettier-ignore
 @function get-color($type, $level) {
-  // add the logic here
-  // use if else to check if the type and level are valid
-  // if they are valid, return the color
-  // if they are not valid warn 'Invalid color type or level' and return null
-  // null is a special value in Sass that means nothing, and it is used to represent the absence of a value, you need to return null if the type or level is not valid
+  @if map.has-key($colors, $type) and map.has-key(map.get($colors, $type), $level) {
+  @return map.get(map.get($colors, $type), $level);
+  } @else {
+
+    @if not map.has-key($colors, $type) {
+      @warn "Invalid color type: #{$type}";
+    }
+
+
+
+    @if map.has-key($colors, $type) and not map.has-key(map.get($colors, $type), $level) {
+    @warn "Invalid level: #{$level} for #{$type}";
+    }
```

Comment:
logic kamu sudah tepat dan baik pada `@if` namun pada bagian `@else` kamu bisa menghapus kedua `@if` di sini. karena kita mengacu pada soal
// if they are not valid warn 'Invalid color type or level' and return null
jadi kamu bisa langsung memunculkan `@warn`nya saja dan `@return null`